core/include/crypto/crypto.h

```diff
@@ -49,15 +49,17 @@
 TEE_Result crypto_init(void);
 
 /* Message digest functions */
-TEE_Result crypto_hash_get_ctx_size(uint32_t algo, size_t *size);
+TEE_Result crypto_hash_alloc_ctx(void **ctx, uint32_t algo);
 TEE_Result crypto_hash_init(void *ctx, uint32_t algo);
 TEE_Result crypto_hash_update(void *ctx, uint32_t algo, const uint8_t *data,
 			      size_t len);
 TEE_Result crypto_hash_final(void *ctx, uint32_t algo, uint8_t *digest,
 			     size_t len);
+void crypto_hash_free_ctx(void *ctx, uint32_t algo);
```

Comment:
`algo` not needed?

Reply:
I'd like it to be implementation defined if it's needed or not.

Reply:
OK makes sense.